src/Hyperion/Benchmark.hs

```diff
@@ -55,7 +60,37 @@ instance Show Benchmark where
   showsPrec x (WithSampling _opt bk) = showsPrec x bk
 
 bench :: String -> Batch () -> Benchmark
-bench name batch = Bench (Text.pack name) batch
+bench name batch = Bench Text.pack (const []) id (const batch) name
+
+benchWith
+  :: (a -> Text)
+  -> (a -> [JSON.Pair])
```

Comment:
I agree it's quite unfortunate. The easiest solution would most likely be to allow the user to return values of some predefined type that we accept are generally compatible with kibana, json, yaml and any other format that we want to use in the future:

``` haskell
data Foo
  = FooText Text
  | FooInt Int
  | FooDouble Double
```

Otherwise we can leave the return type completely abstract and parametrize `Benchmark` over that.